base/compiler/ssair/irinterp.jl

```diff
@@ -255,6 +255,8 @@ function _ir_abstract_constant_propagation(interp::AbstractInterpreter, irsv::IR
 
     (; ir, tpdum, ssa_refined) = irsv
 
+    @assert isempty(ir.new_nodes) "IRCode should be compacted before irinterp"
```

Comment:
@Keno Do you think it is reasonable to put this assertion here? If so, I believe we don't need to pass `idx` to `scan!` callbacks since `inst.idx === idx` would always hold then.

Reply:
I think this is fine.